src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/internal/core/java/codeaction/CodeActionHandler.java

```diff
@@ -173,37 +178,44 @@ public List<? extends CodeAction> codeAction(MicroProfileJavaCodeActionParams pa
 	 * @return the given unresolved CodeAction with the workspace edit resolved
 	 */
 	public CodeAction resolveCodeAction(CodeAction unresolved, IPsiUtils utils) {
-		CodeActionResolveData data = (CodeActionResolveData) unresolved.getData();
-		String participantId = data.getParticipantId();
-		String uri = data.getDocumentUri();
+		try {
+			CodeActionResolveData data = (CodeActionResolveData) unresolved.getData();
+			String participantId = data.getParticipantId();
+			String uri = data.getDocumentUri();
 
-		PsiFile unit = utils.resolveCompilationUnit(uri);
-		if (unit == null) {
-			return null;
-		}
+			PsiFile unit = utils.resolveCompilationUnit(uri);
+			if (unit == null) {
+				return null;
+			}
 
-		int start = DiagnosticsHelper.getStartOffset(unit, data.getRange(), utils);
-		int end = DiagnosticsHelper.getEndOffset(unit, data.getRange(), utils);
-
-		var params = new MicroProfileJavaCodeActionParams();
-		params.setContext(new CodeActionContext(
-				unresolved.getDiagnostics() == null ? Collections.emptyList() : unresolved.getDiagnostics()));
-		params.setResourceOperationSupported(data.isResourceOperationSupported());
-		params.setCommandConfigurationUpdateSupported(data.isCommandConfigurationUpdateSupported());
-		params.setRange(data.getRange());
-		params.setTextDocument(new VersionedTextDocumentIdentifier(uri, null));
-
-		/*JavaCodeActionResolveContext context = new JavaCodeActionResolveContext(unit, start, end - start, utils, params,
-				unresolved);*/
-		JavaCodeActionResolveContext context = new JavaCodeActionResolveContext(unit,
-				start, end - start, utils, params, unresolved);
-		context.setASTRoot(getASTRoot(unit));
-
-		IJavaCodeActionParticipant participant = JavaCodeActionDefinition.EP.extensions()
-				.filter(definition -> unresolved.getKind().startsWith(definition.getKind()))
-				.filter(definition -> participantId.equals(definition.getParticipantId()))
-				.findFirst().orElse(null);
-		return participant.resolveCodeAction(context.copy());
+			utils = utils.refine(utils.getModule(uri));
```

Comment:
required so that https://github.com/redhat-developer/intellij-quarkus/pull/822/files#diff-0183e50c392af65be0cde7d0481bd7bbe78ec8a031844caccec73ef1bc7c6139R114 doesn't throw an NPE